src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LspServices.cs

```diff
@@ -54,5 +54,9 @@ public bool SupportsGetRegisteredServices()
 
     public void Dispose()
     {
+        if (_serviceProvider is IDisposable disposable)
+        {
+            disposable.Dispose();
```

Comment:
`IServiceProvider` isn't IDisposable, but the implementation might be (and is in our case). Without this some things like `FileSystemWatcher` may continue to operate under the covers because they never got disposed.

Reply:
Nit: `(_serviceProvider as IDisposable)?.Dispose();` if you like one-liners

Reply:
Good catch!